### PR TITLE
Replace assert_shutdown with sleep for Xen

### PR DESCRIPTION
**Pull request by mnowaksuse**
This is to workaround poo#16418.

### CODE REVIEW

File: lib/utils.pm [hunk 1]
```diff
@@ -636,7 +636,10 @@ sub assert_shutdown_and_restore_system {
     $shutdown_timeout //= 60;
     my $vnc_console = get_required_var('SVIRT_VNC_CONSOLE');
     console($vnc_console)->disable_vnc_stalls;
-    assert_shutdown($shutdown_timeout);
+    # Ideally, `assert_shutdown` is here instead of `sleep`, but poo#16418
+    # prevent us to be it so. Rather wait couple of seconds more than risk
+    # an 2 hour hang.
+    sleep($shutdown_timeout);
```
Comment (okurz) [inline lib/utils.pm]: well I am not sure if this is a clean solution. Not that "just sleeping" is wrong but doing it here. See https://github.com/os-autoinst/os-autoinst/blob/master/testapi.pm#L1524 , the test API already does the same if the corresponding specific backend does not support it. If we have the feeling that getting changes into os-autoinst takes too long vs. the tests then we should try to change that in the long run but of course on short term it helps when we can do the corresponding changes within the tests distribution. We could try to be more explicit and *monkey patch* the backend functionality by overwriting the backend package implementation within here.

Comment (asmorodskyi) [reply lib/utils.pm]: +1 . I would just phrase this in little bit different way - if some functionality often need to be changed from certain test perceptive probably it's subject to discuss with openQA devs to just move it to test repository ? Or wrap it with   additional layer in test side in way that test developers can fully control it

Comment (okurz) [reply lib/utils.pm]: You can always fully control it. There is no "final" in perl ;) You can overwrite everything. That is called *monkey patching*

Comment (mnowaksuse) [reply lib/utils.pm]: Wasn't my intention to "speed-up os-autoinst" (this time), just was not aware of `is_shutdown` at that moment.

**Pull request merged**